docs/integrations/r.md

```diff
@@ -0,0 +1,177 @@
+* For working with **lakeFS operations such as branches and commits** use the [API](https://docs.lakefs.io/reference/api.html) for which can be accessed from R using the `httr` library. 
+
+_To see examples of R in action with lakeFS please visit the [lakeFS-samples](https://github.com/treeverse/lakeFS-samples/) repository and the [sample](https://github.com/treeverse/lakeFS-samples/blob/main/00_notebooks/R.ipynb) [notebooks](https://github.com/treeverse/lakeFS-samples/blob/main/00_notebooks/R-weather.ipynb)_.
+
+{% include toc.html %}
+
+## Reading and Writing from lakeFS with R
+
+Working with data stored in lakeFS from R is the same as you would with an S3 bucket, via the [S3 Gateway that lakeFS provides](https://docs.lakefs.io/understand/architecture.html#s3-gateway).
+
+You can use any library that interfaces with S3. In this example we'll use the [aws.s3](https://github.com/cloudyr/aws.s3/tree/master) library.
+
+```r
+install.packages(c("aws.s3"))
+```
+
+### Configuration 
+
+The [R S3 client documentation](https://github.com/cloudyr/aws.s3/blob/master/man/s3HTTP.Rd) includes full details of the configuration options available. A good approach for using it with lakeFS set the endpoint and authentication details as environment variables: 
+
+```r
+Sys.setenv("AWS_ACCESS_KEY_ID" = "AKIAIOSFODNN7EXAMPLE",
+           "AWS_SECRET_ACCESS_KEY" = "wJalrXUtnFEMI/K7MDENG/bPxRfiCYEXAMPLEKEY",
+           "AWS_S3_ENDPOINT" = "lakefs.mycorp.com:8000")
+```
+
+In conjunction with this you must also specify `region` and `use_https` _in each call of an `aws.s3` function_ as these cannot be set globally. For example: 
+
+```r
+bucketlist(
+    region="",
+    use_https="false"
+    )
+```
+
+* `region` should always be empty
+* `use_https` should be set to `true` or `false` depending on whether your lakeFS endpoint uses HTTPS.
+
+### Listing repositories
+
+The S3 gateway exposes a repository as a bucket, and so using the `aws.s3` function `bucketlist` will return a list of available repositories on lakeFS: 
+
+```r
+bucketlist(
+    region="",
+    use_https="false"
+    )
+```
+
+### Writing to lakeFS from R
+
+Assuming you're using the `aws.s3` library there various functions available including `s3save`, `s3saveRDS`, and `put_object`. Here's an example of writing an R object to lakeFS: 
+
+```r
+repo_name <- "example"
+branch <- "development"
+
+s3saveRDS(x=my_df, 
+          bucket = repo_name, 
+          object = paste0(branch,"/my_df.R"), 
+          region="",
+          use_https="false")
+```
+
+You can also upload local files to lakeFS using R and the `put_object` function: 
+
+```r
+repo_name <- "example"
+branch <- "development"
+local_file <- "/tmp/never.gonna"
+
+put_object(file = local_file, 
+           bucket = repo_name, 
+           object = paste0(branch,"/give/you/up"),
+           region="",
+           use_https=useHTTPS)
```

Comment:
Think we need to explain `useHTTPS` before we use it - or default all the examples to use `FALSE`. Usually users will just copy/paste and will fail on unknown variables.

```suggestion
           region = "",
           use_https = useHTTPS)
```

Reply:
good point - lazy copypasta from me. fixed.